lib/GenericRepository.ts

```diff
@@ -125,14 +126,23 @@ export abstract class GenericRepository<T extends DecoratedEntity> {
     }
 
     /**
-     * @description Deletes the entity by a given options.
+     * @description Soft-deletes the entity by a given options.
```

Comment:
Change this to `softDelete` while keeping the `hardDelete` method as the default delete operation.